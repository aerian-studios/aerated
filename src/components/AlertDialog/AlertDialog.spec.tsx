import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AlertDialog, AlertDialogFooter, AlertDialogTrigger } from "./index";

describe("AlertDialog", () => {
  it("should render correctly", async () => {
    const { container } = render(
      <AlertDialog
        trigger={
          <AlertDialogTrigger>
            <button>Open the dialog</button>
          </AlertDialogTrigger>
        }
        dialogTitle={<>This is important!</>}
        dialogFooter={
          <AlertDialogFooter
            dialogCancel={<button>Cancel</button>}
            dialogAction={<button>Yes, do it!</button>}
          />
        }
        dialogContent={<p>Are you sure about all of that stuff?</p>}
      />
    );

    await userEvent.click(
      screen.getByRole("button", { name: "Open the dialog" })
    );

    expect(screen.getByRole("alertdialog", { name: "This is important!" }));
    expect(screen.getByText("Are you sure about all of that stuff?"));

    expect(container).toMatchSnapshot();
  });
});
