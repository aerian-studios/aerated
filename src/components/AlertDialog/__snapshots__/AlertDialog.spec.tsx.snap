// Vitest Snapshot v1

exports[`AlertDialog > should render correctly 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="true"
    aria-haspopup="dialog"
    data-state="open"
    type="button"
  >
    Open the dialog
  </button>
</div>
`;
